Azure.ResourceManager.Core.Tests/ResourceIdentifierTests.cs

```diff
@@ -18,6 +18,41 @@ public void Setup()
         {
         }
 
+        [TestCase("")]
+        [TestCase(" ")]
+        [TestCase("asdfghj")]
+        [TestCase("123456")]
+        [TestCase("!@#$%^&*/")]
+        [TestCase("/subscriptions/")]
+        [TestCase("/0c2f6471-1bf0-4dda-aec3-cb9272f09575/myRg/")]
+        public void InvalidRPIds(string invalidID)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => { ResourceIdentifier subject = invalidID; });
+        }
+
+        [TestCase (null)]
+        [TestCase (TrackedResourceId)]
+        public void ImplicitConstructor(string resourceProviderID)
+        {
+            string x = resourceProviderID;
+            string y;
+            ResourceIdentifier z;
+
+            z = x;
+            y = z;
+
+            if (resourceProviderID is null)
+            {
+                Assert.IsNull(z);
+                Assert.IsNull(y);
+            }
+            else
+            {
+                Assert.AreEqual(resourceProviderID, y);
+            }
+            
+        }
+
```

Comment:
add a new test case which validates passing in null to the public constructor

Reply:
Doesn't look like this got added

Reply:
It was added in the last commit